crates/re_space_view/src/data_query_blueprint.rs

```diff
@@ -384,25 +244,25 @@ impl<'a> QueryExpressionEvaluator<'a> {
         overrides: &EntityOverrides,
         inherited: &EntityProperties,
         data_results: &mut SlotMap<DataResultHandle, DataResultNode>,
-        from_recursive: bool,
     ) -> Option<DataResultHandle> {
         // If we hit a prefix that is not allowed, we terminate. This is
-        // a pruned branch of the tree. Can come from either an explicit
-        // recursive exclusion, or an implicit missing inclusion.
+        // a pruned branch of the tree.
+
+        // TODO(emilk): implement this optimization
+        // if self
+        //     .entity_path_filter
+        //     .is_everything_starting_with_excluded(&tree.path)
+        // {
+        //     return None;
+        // }
+
         // TODO(jleibs): If this space is disconnected, we should terminate here
-        if self.recursive_exclusions.contains(&tree.path)
-            || !(from_recursive || self.allowed_prefixes.contains(&tree.path))
-        {
-            return None;
-        }
 
         let entity_path = tree.path.clone();
 
         // Pre-compute our matches
-        let exact_include = self.exact_inclusions.contains(&entity_path);
-        let recursive_include = self.recursive_inclusions.contains(&entity_path) || from_recursive;
-        let exact_exclude = self.exact_exclusions.contains(&entity_path);
-        let any_match = (exact_include || recursive_include) && !exact_exclude;
+        let exact_include = self.entity_path_filter.is_exact_included(&entity_path);
+        let any_match = self.entity_path_filter.is_included(&entity_path);
```

Comment:
Nice -- this shouldn't conflict too much with my other open-PR for query perf stuff.